components/ViewHabits.tsx

```diff
@@ -84,16 +180,42 @@ const ViewHabits = () => {
       </header>
 
       <div css={styles.habitsContainer}>
-        {habitsData.map(({ habits }) =>
-          habits.map(({ entries, streak, ...habit }) => (
-            <Habit
-              key={habit.id}
-              entries={entries}
-              streak={streak}
-              habit={habit}
-            />
-          ))
+        {timeView === 'weekly' && (
+          <table css={styles.weeklyContainer}>
+            <thead>
+              <tr>
+                <th>Habit name</th>
+                {dates.map((date) => (
+                  <th key={date.toDateString()}>
+                    {date.toLocaleDateString('en-GB', { weekday: 'narrow' })}{' '}
+                    {date.getDate()}
+                  </th>
+                ))}
+              </tr>
+            </thead>
+
+            {habitsData.map(({ habits }) =>
+              habits.map((habitWithHistory) => (
+                <HabitWeeklyView
+                  key={habitWithHistory.id}
+                  dates={dateStrings}
+                  habitWithHistory={habitWithHistory}
+                />
+              ))
+            )}
+          </table>
```

Comment:
These presentations feel like different components, still. Is there no way to move this table into `HabitWeeklyView?` That way, you'd be able to do the mapping once.

Additionally, since the `habits` folder doesn't appear to expose an entrypoint of sorts (index.tsx) - it's a little confusing to have it at as a folder there - maybe it should be colocated with `ViewHabits`, as a subcomponent/in the same directory there OR it should expose an importable entrypoint component. Do also change the name to PascalCase when you do this!